pkgs/sdk/vm/keeper.go

```diff
@@ -130,15 +131,47 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 	if creatorAcc == nil {
 		return std.ErrUnknownAddress(fmt.Sprintf("account %s does not exist", creator))
 	}
+
 	if err := msg.Package.Validate(); err != nil {
 		return ErrInvalidPkgPath(err.Error())
 	}
 	if pv := store.GetPackage(pkgPath, false); pv != nil {
 		// TODO: return error instead of panicking?
 		panic("package already exists: " + pkgPath)
 	}
+
 	// Pay deposit from creator.
 	pkgAddr := gno.DerivePkgAddr(pkgPath)
+
+	// only realm owner can add realm subpackages.
+	if gno.IsRealmPath(pkgPath) {
+		pathSp := strings.SplitN(pkgPath, "/", 4) // gno.land/r/...
+		rootPath := strings.Join(pathSp[0:3], "/")
+		if pkgPath != rootPath {
+			if pv := store.GetPackage(rootPath, false); pv == nil {
+				// TODO: return error instead of panicking?
+				panic("root package doesn't exist: " + rootPath)
+			}
+
+			rootAcc := vm.acck.GetAccount(ctx, gno.DerivePkgAddr(rootPath))
+			if rootAcc == nil {
+				panic("root package account doesn't exist: " + rootPath)
+			}
+
+			if creator != rootAcc.(*types.GnoAccount).Owner {
+				panic("only root realm package creator can create realm subpackages ")
+			}
+		}
+	}
+
+	// Add package account.
+	pkgAcc := vm.acck.NewAccountWithAddress(ctx, pkgAddr)
+	ga := pkgAcc.(*types.GnoAccount)
+	ga.PackageAccount = &types.PackageAccount{
+		Owner: creator,
+	}
+	vm.acck.SetAccount(ctx, ga)
```

Comment:
What do you think about removing this part and make the `creator != rootAcc.Owner` recompute the parent realm’s author dynamically?